File: java-example/src/main/java/org/onflow/examples/java/getExecutionData/GetExecutionDataAccessAPIConnector.java
package org.onflow.examples.java.getExecutionData;

import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowExecutionResult;
import org.onflow.flow.sdk.FlowId;

public class GetExecutionDataAccessAPIConnector {
    private final FlowAccessApi accessAPI;

    public GetExecutionDataAccessAPIConnector(FlowAccessApi accessAPI) {
        this.accessAPI = accessAPI;
    }

    public FlowExecutionResult getExecutionDataByBlockId(FlowId blockId) {
        FlowAccessApi.AccessApiCallResponse<FlowExecutionResult> response = accessAPI.getExecutionResultByBlockId(blockId);
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowExecutionResult>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }
}

File: kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/getExecutionData/GetExecutionDataAccessAPIConnector.kt
package org.onflow.examples.kotlin.getExecutionData

import org.onflow.flow.sdk.*

internal class GetExecutionDataAccessAPIConnector(private val accessAPI: FlowAccessApi) {

    fun getExecutionDataByBlockId(blockId: FlowId): FlowExecutionResult {
        return when (val response = accessAPI.getExecutionResultByBlockId(blockId)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }
}

